src/main/java/org/stellar/sdk/requests/SSEManager.java

```diff
@@ -0,0 +1,81 @@
+package org.stellar.sdk.requests;
+
+import okhttp3.OkHttpClient;
+import okhttp3.sse.EventSource;
+
+import java.io.Closeable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+interface CloseListener {
+  public void closed(EventSource source);
+}
+
+public class SSEManager<T extends org.stellar.sdk.responses.Response> implements Closeable {
```

Comment:
Let me know what do you think about this:

This class represents a single stream so probably `SSEStream` is a better name. When it comes to a single stream, I think it should only have a single method: `close()` that developer can call when stream is no longer needed. The stream should make use of `id` field in a stream and use this in `Last-Event-ID` request header when making a new connection after Horizon closes the connection. If the response code is not in `2xx` range or stream sends `event: err` event, it should stop and call `onError` listener.

This is defined in [SSE spec](https://www.w3.org/TR/2009/WD-eventsource-20090421/).

Reply:
I renamed the class. okhttp-sse follows the spec and has appropriate listener methods. I've extracted the anonymous class for this to a private static inner class so it is a bit easier to follow. Since we use okhttp-sse, I don't see how we can easily get to the last-event-id unless it already does this for us somehow.

But we are tracking the pagingToken that comes back with each event. I've added that to SSEStream with an atomic reference so you can find out which pagingToken was last seen.

Reply:
correction, just noticed that the listener actually passes the id. I now track that and set the header.